app/services/mentor_at_school_periods/change_lead_provider.rb

```diff
@@ -0,0 +1,112 @@
+module MentorAtSchoolPeriods
+  class ChangeLeadProvider
+    class LeadProviderNotChangedError < StandardError; end
+
+    attr_reader :mentor_at_school_period,
+                :lead_provider,
+                :author
+
+    include TrainingPeriodSources
+
+    def initialize(mentor_at_school_period:, lead_provider:, author:)
+      @mentor_at_school_period = mentor_at_school_period
+      @lead_provider = lead_provider
+      @author = author
+    end
+
+    def call
+      raise LeadProviderNotChangedError unless lead_provider_changed?
```

Comment:
> I am not sure if I am dealing correctly with the situation where the Lead Provider hasn't changed. Currently the service raises an error, which is caught by the wizard. The wizard has validation for lead provider not changing, but I did wonder whether that was sufficient to stop a rogue request going through. Unlike the service to change LP for an ECT, afaik it is not possible for the mentor to have school led training, so I do not raise an error for that situation

I think this is fine, you're right, mentors are always provider-led. The unchanged LP check already covers the only invalid state that might hit. It's nice defensive programming IMO.